signer/src/api/info.rs

```diff
@@ -0,0 +1,505 @@
+                    return;
+                };
+
+                self.bitcoin.signer_tip = Some(ChainTipInfo {
+                    block_hash: bitcoin_block.block_hash,
+                    block_height: bitcoin_block.block_height,
+                });
+
+                match storage
+                    .get_stacks_chain_tip(&bitcoin_block.block_hash)
+                    .await
+                {
+                    Ok(Some(local_stacks_chain_tip)) => {
+                        self.stacks.signer_tip = Some(ChainTipInfo {
+                            block_hash: local_stacks_chain_tip.block_hash,
+                            block_height: local_stacks_chain_tip.block_height,
+                        });
+                    }
+                    Ok(None) => {
+                        tracing::debug!("no local stacks tip found in the database.");
+                    }
+                    Err(e) => {
+                        tracing::error!("error reading local Stacks tip from the database: {}", e);
+                    }
+                }
+            }
+            Ok(None) => {
+                tracing::debug!("no local bitcoin tip found in the database.");
+            }
+            Err(e) => {
+                tracing::error!("error reading bitcoin tip from the database: {}", e);
+            }
+        }
+    }
+
+    /// Populates the Bitcoin node tip information from the provided Bitcoin
+    /// client. This uses a combination of `getblockchaininfo` and
+    /// `getnetworkinfo` RPC calls to populate the information.
+    async fn populate_bitcoin_node_info(&mut self, bitcoin_client: &impl BitcoinInteract) {
+        match bitcoin_client.get_blockchain_info().await {
+            Ok(info) => {
+                self.bitcoin.node_chain = Some(info.chain.to_string());
+                self.bitcoin.node_tip = Some(ChainTipInfo {
+                    block_hash: info.best_block_hash.into(),
+                    block_height: info.blocks,
+                });
+            }
+            Err(e) => {
+                tracing::error!("error getting bitcoin node blockchain info: {}", e);
+            }
+        }
+
+        match bitcoin_client.get_network_info().await {
+            Ok(info) => {
+                self.bitcoin.node_version = Some(info.version);
+                self.bitcoin.node_subversion = Some(info.subversion);
+            }
+            Err(e) => {
+                tracing::error!("error getting bitcoin node network info: {}", e);
+            }
+        }
+    }
+
+    /// Populates the Stacks node tip information from the provided Stacks client.
+    /// This uses the `/v2/info` RPC endpoint to populate the information.
+    async fn populate_stacks_node_info(&mut self, stacks_client: &impl StacksInteract) {
+        match stacks_client.get_node_info().await {
+            Ok(node_info) => {
+                self.stacks.node_tip = Some(ChainTipInfo {
+                    block_hash: node_info.stacks_tip.0.into(),
```

Comment:
If this is the block hash and not the block id it could explain the mismatch you mentioned in the PR

Reply:
Yeah, I assume it's something like that... I might be able to get it if I add a new Stacks RPC that can search for the block (iirc they have some search criteria that might work).

Right now I feel pretty content with at least having the block height and adding that kind of improvement later vs. spending time on it now.

Reply:
Yeah makes sense, also knowing the difference it's "easy" to crosscheck anyway. Maybe we could rename `block_hash` under `signer_tip` to make it clear that is referring to the block index hash.

Reply:
Yeah that works, updated in [6b19e02](https://github.com/stacks-network/sbtc/pull/1251/commits/6b19e0241bd2f88e178927bbe53c24e763ea77c2)